### PR TITLE
Fix #854 - PowerShell aliases against app with .exe ext doesn't work

### LINKED ISSUE #854: Tab completion of a PowerShell alias does not work if alias was defined using app name with .exe extension
### System Details

- posh-git version/path: 1.0.0  ~\Documents\PowerShell\Modules\posh-git\1.0.0
- PowerShell version: 7.1.3
- git version 2.28.0.windows.1
- OS: Microsoft Windows NT 10.0.19042.0

### Issue Description

Title pretty much says it all.  Basically, the following won't work:
```powershell
New-Alias g git.exe
```
Whereas this does work:
```powershell
New-Alias g git
```

See also #852

### PR DESCRIPTION
**Pull request by rkeithhill**
Add RegisteredCommands field to GitTabSettings for debug purposes.

### CODE REVIEW

File: src/GitTabExpansion.ps1 [hunk 1]
```diff
@@ -537,10 +538,21 @@ function WriteTabExpLog([string] $Message) {
 
 if (!$UseLegacyTabExpansion -and ($PSVersionTable.PSVersion.Major -ge 6)) {
     $cmdNames = "git","tgit","gitk"
+
+    # Create regex pattern from $cmdNames: ^(git|git\.exe|tgit|tgit\.exe|gitk|gitk\.exe)$
+    $cmdNamesPattern = "^($(($cmdNames | ForEach-Object { "${_}|${_}\.exe" }) -join '|'))$"
```
Comment (dahlbyk) [inline src/GitTabExpansion.ps1:543]: Simpler to just optionally match `.exe` at the end?
```suggestion
    $cmdNamesPattern = "^($($cmdNames -join '|'))(\.exe)?$"
```

File: src/GitUtils.ps1 [hunk 2]
```diff
@@ -442,7 +442,7 @@ function InDotGitOrBareRepoDir([string][ValidateNotNullOrEmpty()]$GitDir) {
 }
 
 function Get-AliasPattern($cmd) {
-    $aliases = @($cmd) + @(Get-Alias | Where-Object { $_.Definition -eq $cmd } | Select-Object -Exp Name)
+    $aliases = @($cmd) + @(Get-Alias | Where-Object { $_.Definition -match "^($cmd|$cmd\.exe)$" } | Foreach-Object Name)
```
Comment (dahlbyk) [inline src/GitUtils.ps1:445]:
```suggestion
    $aliases = @($cmd) + @(Get-Alias | Where-Object { $_.Definition -match "^$cmd(\.exe)?$" } | Foreach-Object Name)
```

File: src/GitTabExpansion.ps1 [hunk 3]
```diff
@@ -537,10 +538,21 @@ function WriteTabExpLog([string] $Message) {
...
+    $cmdNames += Get-Alias | Where-Object { $_.Definition -match $cmdNamesPattern } | Foreach-Object Name
+
     if ($EnableProxyFunctionExpansion) {
-        $cmdNames += Get-ChildItem -Path Function:\ | Where-Object { $_.Definition -match $script:GitProxyFunctionRegex } | Select-Object -ExpandProperty 'Name'
+        $funcNames += Get-ChildItem -Path Function:\ | Where-Object { $_.Definition -match $script:GitProxyFunctionRegex } | Foreach-Object Name
+        $cmdNames += $funcNames
+
+        # Create regex pattern from $funcNames e.g.: ^(Git-Checkout|Git-Switch)$
+        $funcNamesPattern = "^($($funcNames -join '|'))$"
+        $cmdNames += Get-Alias | Where-Object { $_.Definition -match $funcNamesPattern } | Foreach-Object Name
```
Comment (dahlbyk) [inline src/GitTabExpansion.ps1:552]: Add test for helper function alias match?

**Review by dahlbyk — APPROVED** [review]
Nothing blocking, but a few questions/ideas.

File: src/GitTabExpansion.ps1 (hunk 1 above)
Comment (rkeithhill) [reply src/GitTabExpansion.ps1]: Done

File: src/GitUtils.ps1 (hunk 2 above)
Comment (rkeithhill) [reply src/GitUtils.ps1]: Done

File: src/GitTabExpansion.ps1 (hunk 3 above)
Comment (rkeithhill) [reply src/GitTabExpansion.ps1:553]: Done

**Comment by dahlbyk** [comment]
Thanks!

**Pull request merged**